newton/_src/utils/selection.py

```
@@ -489,73 +652,86 @@ def is_contiguous_slice(indices):
+        )
 
         if _slice is None:
-            _slice = self._frequency_slices.get(frequency)
+            _slice = layout.slice
         elif isinstance(_slice, Slice):
             _slice = _slice.get()
         elif isinstance(_slice, int):
-            return attrib[:, _slice]
+            _slice = slice(_slice, _slice + 1)
         else:
             raise TypeError(f"Invalid slice type: expected Slice or int, got {type(_slice)}")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)